File: backend/app/schemas/osint.py
from pydantic import BaseModel, Field


class OsintEnrichRequest(BaseModel):
    case_id: str
    entity_id: str
    lookup_id: str


class OsintEnrichmentHit(BaseModel):
    title: str
    detail: str
    source_registry: str
    relevance_score: int = 0
    match_quality: str = "none"
    simulated: bool = True
    source_type: str = "suggested"
    requires_officer_review: bool = False


class OsintEnrichResponse(BaseModel):
    enrichment_id: str
    entity_id: str
    lookup_id: str
    lookup_label: str
    case_id: str
    simulated: bool = True
    disclaimer: str = ""
    results: list[OsintEnrichmentHit] = Field(default_factory=list)
    graph_links_added: list[str] = Field(default_factory=list)
    suggested_links: list[str] = Field(default_factory=list)
    audit_entry_id: str
    audit_hash: str


class AuditChainEntry(BaseModel):
    id: str
    timestamp: str
    action: str
    entityId: str
    source: str
    operator: str
    lawfulBasis: str
    entryHash: str
    prevHash: str


class OsintAuditLogResponse(BaseModel):
    case_id: str | None = None
    entries: list[AuditChainEntry] = Field(default_factory=list)
    chain_status: str = "unknown"


class AuditVerifyResponse(BaseModel):
    status: str
    entries_checked: int
    broken_entry_id: str | None = None
    message: str

File: frontend/src/api/osint.ts
import { apiGet, apiPost, isApiConfigured } from './client'
import type { AuditEntry } from '../types'

export interface OsintEnrichRequest {
  case_id: string
  entity_id: string
  lookup_id: string
}

export interface OsintEnrichmentHit {
  title: string
  detail: string
  source_registry: string
  relevance_score: number
  match_quality: string
  simulated: boolean
  source_type: string
  requires_officer_review: boolean
}

export interface OsintEnrichResponse {
  enrichment_id: string
  entity_id: string
  lookup_id: string
  lookup_label: string
  case_id: string
  simulated: boolean
  disclaimer: string
  results: OsintEnrichmentHit[]
  graph_links_added: string[]
  suggested_links: string[]
  audit_entry_id: string
  audit_hash: string
}

export interface OsintAuditLogResponse {
  case_id: string | null
  entries: (AuditEntry & { entryHash?: string; prevHash?: string })[]
  chain_status: string
}

export interface AuditVerifyResponse {
  status: string
  entries_checked: number
  broken_entry_id: string | null
  message: string
}

export async function enrichEntity(
  payload: OsintEnrichRequest,
): Promise<OsintEnrichResponse> {
  return apiPost<OsintEnrichResponse>('/osint/enrich', payload)
}

export async function getOsintAuditLog(caseId: string): Promise<OsintAuditLogResponse> {
  return apiGet<OsintAuditLogResponse>(`/osint/audit-log?case_id=${encodeURIComponent(caseId)}`)
}

export async function verifyAuditChain(caseId?: string): Promise<AuditVerifyResponse> {
  const qs = caseId ? `?case_id=${encodeURIComponent(caseId)}` : ''
  return apiGet<AuditVerifyResponse>(`/audit/verify${qs}`)
}

export function isOsintLive(): boolean {
  return isApiConfigured()
}
